presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -88,12 +88,14 @@ public synchronized ScheduleResult schedule()
                     // no more splits
                     switch (state) {
                         case STARTED:
-                            // we have not scheduled a single split so far, schedule empty split
-                            state = State.SCHEDULED;
-                            Set<RemoteTask> emptyTask = assignSplits(ImmutableMultimap.of(
-                                    splitPlacementPolicy.anyNode(),
-                                    splitSource.getEmptySplit()));
-                            return new ScheduleResult(false, emptyTask, 1);
+                            if (stage.getFragment().hasDefaultOutput()) {
```

Comment:
this commit seems to be like a bugfix, so this commit should be a least merged with the commit which introduced scheduling empty split